src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -65,13 +93,49 @@ public MesosCloud(String name) throws InterruptedException, ExecutionException {
    *
    * @return A future reference to the launched node.
    */
-  private CompletableFuture<Node> startAgent() {
-    // mesos.enqueueAgent().thenCompose(mesosSlave -> mesosSlave.waitUntilOnlineAsync());
-    throw new NotImplementedException();
-  }
-
   @Override
   public boolean canProvision(Label label) {
-    throw new NotImplementedException();
+    // TODO: implement executor limits
+    return true;
+  }
+
+  public MesosApi getMesosClient() {
+    return this.mesos;
+  }
+
+  /**
+   * Start a Jenkins agent.jar on Mesos.
+   *
+   * <p>Provide a callback for Jenkins to start a Node.
+   *
+   * @return A future reference to the launched node.
+   */
+  private class ProvisioningCallback implements Callable<Node> {
+    MesosCloud cloud;
+
+    ProvisioningCallback(MesosCloud cloud) {
+      this.cloud = cloud;
+    }
+
+    @Override
+    public Node call() throws Exception {
+      return mesos
+          .enqueueAgent(cloud, 0.1, 32)
+          .thenApply(
+              mesosSlave -> {
+                Future<MesosSlave> agent = null;
+                try {
+                  Jenkins.getInstanceOrNull().addNode(mesosSlave);
+                  LOGGER.info("waiting for slave to come online...");
+                  agent = mesosSlave.waitUntilOnlineAsync();
```

Comment:
I think this should be a `thenCompose` since `agent` is a future. Why are we initializing the agent with `null`? I don't think that's required. As a follow up we could simplify thinks with [ea-await](https://github.com/electronicarts/ea-async).

Reply:
ya was using the null as a test check, fixed this up

Reply:
switched the thenCompose had to make some changes in mesosSlave